crates/rune/src/query/mod.rs

```diff
@@ -1633,18 +1633,30 @@ fn tuple_body_meta(
 
 /// Construct metadata for a struct body.
 fn struct_body_meta(
-    item: &Item,
+    item: &Arc<ItemMeta>,
     enum_: Option<(&Item, Hash, usize)>,
     ctx: ResolveContext<'_>,
     st: ast::Braced<ast::Field, T![,]>,
 ) -> Result<PrivMetaKind, QueryError> {
-    let type_hash = Hash::type_hash(item);
-
-    let mut fields = HashSet::new();
-
-    for (ast::Field { name, .. }, _) in st {
-        let name = name.resolve(ctx)?;
-        fields.insert(name.into());
+    let type_hash = Hash::type_hash(&item.item);
+
+    let mut fields = HashMap::new();
+
+    for (field, _) in st {
+        let mut buf = ItemBuf::with_item(&item.item);
+        let name = field.name.resolve(ctx)?;
+        buf.push(name);
+
+        fields.insert(
+            name.into(),
+            Arc::new(ItemMeta {
+                id: Id::default(),
+                location: Location::new(item.location.source_id, field.span()),
+                item: buf,
+                visibility: Visibility::from_ast(&field.visibility).unwrap(),
```

Comment:
Need a way to avoid this unwrap.

Reply:
This was replaced by using `QueryError::msg`, though I wonder if it's possible to even encounter this error at this stage.